pep-0646.rst

```diff
@@ -414,6 +354,133 @@ Normal ``TypeVar`` instances can also be prefixed and/or suffixed:
 
+Unpacking a Tuple Type
+----------------------
+
+We mentioned that a ``TypeVarTuple`` simply stands for a tuple of
+types. Thus, we can cleanly replace any use of a ``TypeVarTuple`` in a
+function signature with a tuple type.
+
+
+As with ``TypeVarTuple``, only one unpacking may appear in a tuple:
+
+::
+
+    x: Tuple[int, *Ts, str, *Ts2]  # Error
+    y: Tuple[int, *Tuple[int, ...], str, *Tuple[str, ...]]  # Error
+
+
+Unpacking a Concrete Tuple Type
+'''''''''''''''''''''''''''''''
+
+Unpacking a concrete tuple type is analogous to unpacking a tuple of
+values at runtime. ``Tuple[int, *Tuple[bool, bool], str]`` is
+equivalent to ``Tuple[int, bool, bool, str]``. While this is unlikely
+to be useful by itself, we mention it to cleanly specify how we unpack
+a ``TypeVarTuple`` that is substituted by a concrete tuple type:
+
+::
+
+    def strip_and_add_float(x: Tuple[int, *Ts, str]) -> Tuple[float, *Ts, float]: ...
+
+When ``strip_and_add_float`` is called with a value that has a concrete
+tuple type, say, ``Tuple[int, bool, bool, str]``, ``Ts`` is bound to
+the concrete tuple ``Tuple[bool, bool]``.
+
+::
+
+    x: Tuple[int, bool, bool, str] = (1, True, False, "hello")
+
+    y = strip_and_add_float(x)
+    # Inferred type of y: Tuple[float, bool, bool, float]
+
+The inferred type is ``Tuple[float, bool, bool, float]`` because
+unpacking ``Tuple[bool, bool]`` in the return type ``Tuple[float, *Ts,
+float]`` gives back ``Tuple[float, bool, bool, float]``.
+
+Unpacking an Unbounded Tuple Type
+'''''''''''''''''''''''''''''''''
+
+Unpacking an unbounded tuple preserves the unbounded tuple as it is.
+``Tuple[int, *Tuple[str, ...], str]`` signifies a tuple
+type where the first element is guaranteed to be of type ``int``, the
+last element is guaranteed to be of type ``str``, and there may be
+zero or more elements of type ``str`` in between. Note that
+``Tuple[*Tuple[int, ...]]`` is equivalent to ``Tuple[int, ...]``.
+
+Consider the following function:
+
+::
+
+    def add_first_last(x: Tuple[*Ts]) -> Tuple[int, *Ts, str]: ...
+
+When ``add_first_last`` is called with a value that has an *unbounded*
+tuple type, say, ``Tuple[str, ...]``, ``Ts`` is bound to the concrete
+tuple ``Tuple[str, ...]`` and substituted in the return type.
+
+::
+
+    x: Tuple[str, ...]
+
+    y = add_first_last(x)
+    # Inferred type of y: Tuple[int, *Tuple[str, ...], str]
+
+
+Unpacking unbounded tuples is useful in function signatures where we
+don't care about the exact elements and don't want to define an
+unnecessary ``TypeVarTuple``:
+
+::
+
+    def process_batch_channels(
+        x: Array[Batch, *Tuple[Any, ...], Channels]
+    ) -> None:
+        ...
+
+
+    x: Array[Batch, Height, Width, Channels]
+    process_batch_channels(x)  # OK
+    y: Array[Batch, Channels]
+    process_batch_channels(y)  # OK
+    z: Array[Batch]
+    process_batch_channels(z)  # Error: Expected Channels.
+
+
+We can also pass a ``Tuple[int, ...]`` wherever a ``Tuple[*Ts]`` is
```

Comment:
I found this first sentence a bit confusing - is it saying that we can pass a ``Tuple[int, ...]`` wherever a ``*Ts`` appears in the argument list?